desktop/core/src/desktop/middleware_test.py

```diff
@@ -201,3 +204,30 @@ def test_spnego_middleware():
       finish()
     settings.AUTHENTICATION_BACKENDS = orig_backends
 
+def test_cache_control_middleware():
+  c = Client()
+  request = c.get("/")
+
+  def dummy_get_response(request):
+    return HttpResponse()
+
+  CUSTOM_CACHE_CONTROL.set_for_testing(True)
+  try:
+    middleware = CacheControlMiddleware(dummy_get_response)
+    response = middleware(request)
+    assert_equal(response['Cache-Control'], 'no-cache, no-store, must-revalidate')
+    assert_equal(response['Pragma'], 'no-cache')
+    assert_equal(response['Expires'], '0')
+  except Exception as e:
+    print("exception occured:", e.exception)
+
+  CUSTOM_CACHE_CONTROL.set_for_testing(False)
+  try:
+    middleware = CacheControlMiddleware(dummy_get_response)
+    response = middleware(request)
+    assert_not_in('Cache-Control', response)
+    assert_not_in('Pragma', response)
+    assert_not_in('Expires', response)
+  except exceptions.MiddlewareNotUsed:
+    response = dummy_get_response(request)
+
```

Comment:
1. Same, add reset try finally block
2. We can try adding an assert statement `assert_equal('Unloading CacheControlMiddleware, str(e)` at/after line 232.

Can use nested try block for the above changes

Reply:
Added reset try finally block. 
exceptions.MiddlewareNotUsed returns a None string. Not able to add an assert_equal()